onnxruntime/core/session/inference_session.cc

```diff
@@ -1578,51 +1590,74 @@ common::Status InferenceSession::Initialize() {
+            if (!res.first) {
+              LOGS(*session_logger_, ERROR) << "This session cannot use the CUDA Graph feature as requested by the user "
+                                            << " as all compute graph nodes have not been partitioned to the CUDA EP.";
 
-          // Return error status as we don't want the session initialization to complete successfully
-          // if the user has requested usage of CUDA Graph feature and we cannot honor that.
-          ORT_RETURN_IF_ERROR_SESSIONID_(
-              ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
-                              "This session cannot use the CUDA Graph feature as requested by the user "
-                              " as all compute graph nodes have not been partitioned to the CUDA EP."));
-        }
+              // Return error status as we don't want the session initialization to complete successfully
+              // if the user has requested usage of CUDA Graph feature and we cannot honor that.
+              ORT_RETURN_IF_ERROR_SESSIONID_(
+                  ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                                  "This session cannot use the CUDA Graph feature as requested by the user "
+                                  " as all compute graph nodes have not been partitioned to the CUDA EP."));
+            }
 
-        if (res.second > 0) {
-          LOGS(*session_logger_, WARNING) << "This model has shape massaging nodes that will execute on CPU. "
-                                          << "Use the CUDA Graph feature with caution. "
-                                          << "As long as the intermediate shapes produced in the model "
-                                          << "using the representative input used to capture the CUDA graph, "
-                                          << "will match the shapes produced in the model for other inputs "
-                                          << "of the same shape as the representative input (common case), "
-                                          << "it is safe to use the CUDA Graph feature.";
-        }
+            if (res.second > 0) {
+              LOGS(*session_logger_, WARNING) << "This model has shape massaging nodes that will execute on CPU. "
+                                              << "Use the CUDA Graph feature with caution. "
+                                              << "As long as the intermediate shapes produced in the model "
+                                              << "using the representative input used to capture the CUDA graph, "
+                                              << "will match the shapes produced in the model for other inputs "
+                                              << "of the same shape as the representative input (common case), "
+                                              << "it is safe to use the CUDA Graph feature.";
+            }
+          } else {
```

Comment:
Nit: Maybe add a comment for this else to say - currently this code path is for the TRT EP?